### PR TITLE
doc: terminology cleanup in mem mgmt hld

### PR DESCRIPTION
**Pull request by amyreye**
- Replace SOS or Service OS with Service VM
- Replace UOS or User OS with User VM
- Clean up some of the grammar

Signed-off-by: Amy Reyes <amy.reyes@intel.com>

### CODE REVIEW

File: doc/developer-guides/hld/hv-memmgt.rst [hunk 1]
```diff
@@ -52,16 +52,18 @@ inside the hypervisor and from a VM:
...
 *************************************
 
 In ACRN, the HV initializes MMU page tables to manage all physical
 memory and then switches to the new MMU page tables. After MMU page
 tables are initialized at the platform initialization stage, no updates
-are made for MMU page tables except when hv_access_memory_region_update is called.
-However, the memory region updated by hv_access_memory_region_update
+are made for MMU page tables except when ``hv_access_memory_region_update`` is
+called.
+However, the memory region updated by ``hv_access_memory_region_update``
```
Comment (amyreye) [inline doc/developer-guides/hld/hv-memmgt.rst:66]: Where is "hv_access_memory_region_update" in the code?

Comment (dbkinder) [reply doc/developer-guides/hld/hv-memmgt.rst:66]: That function existed in mmu.h in v2.2 but not in v2.3, and it looks like it was renamed to ``ppt_clear_user_bit`` as indicated in this commit title:
```
f3067f538 hv: mmu: rename hv_access_memory_region_update to ppt_clear_user_bit
```
But now that function doesn't seem to exist any more either; it disappeared with v2.5.  @NanlinXie can you help figure out how to update this documentation that mentions ``hv_access_memory_region_update``?

Comment (NanlinXie) [reply doc/developer-guides/hld/hv-memmgt.rst:66]: @lifeix , please help to update HLD.

Comment (lifeix) [reply doc/developer-guides/hld/hv-memmgt.rst:66]: @amyreye  Would you please help to the below update:
-are made for MMU page tables except when hv_access_memory_region_update is called.
-However, the memory region updated by hv_access_memory_region_update
+are made for MMU page tables except when set_paging_supervisor/nx/x is called.
+However, the memory region updated by set_paging_supervisor/nx/x

**Review by lifeix — CHANGES_REQUESTED** [review]
Please replace "hv_access_memory_region_update" with "set_paging_supervisor/nx/x"

**Comment by amyreye** [comment]
> Please replace "hv_access_memory_region_update" with "set_paging_supervisor/nx/x"

Done

**Review by lifeix — APPROVED** [review]

**Comment by amyreye** [comment]
@wenlingz Please rerun the checks. Let's see if it passes now.

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

**Comment by amyreye** [comment]
@wenlingz Doc check is successful. Can you remove "apply fail" tag?

**Review by gvancuts — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
LGTM, thanks

**Pull request merged**